app/api/slang.py

```diff
@@ -17,30 +19,50 @@ def generate_slang():
     return ret
 
 
-def generate_slang_internal():
+def generate_slang_internal(model_type=None):
     """Implement internal call to generate and return a new slang word instead of API call."""
+
+    # Functionality to convert model selection to filename for load_model.
+    model_filename_dict = {
+        "Straattaal": "2021_straattaal_epoch100.pt",
+        "Plaatsnamen": "plaatsnamen_epoch10.pt",
+        "Nederlandse woorden": "pretrained_dutch_epoch3.pt",
+        "Familienamen": "familienamen_epoch4.pt",
+    }
+    vocab_filename_dict = defaultdict(lambda: "vocabulary.txt")
+    vocab_filename_dict["Familienamen"] = "vocabulary_extended.txt"
+
+    # Up to interpretation: which do we consider "existing"?
+    existing_dict = defaultdict(
+        lambda: ["straattaal.txt", "dutch.txt", "plaatsnamen.txt"]
+    )
+    existing_dict["Familienamen"] = ["familienamen.txt"]
+
     # Retrieve model from the session if it's already stored there.
     if not (
-        session.get("model", None)
-        and session.get("vocab", None)
+        session.get("model" + model_type, None)
+        and session.get("vocab" + model_type, None)
         and session.get("dataset", None)
     ):
         print("Model isn't stored in session! Loading...")
-        model, vocab = load_model()
-        session["model"] = model
-        session["vocab"] = vocab
+        model, vocab = load_model(
+            model_filename_dict[model_type],
+            filename_vocab=vocab_filename_dict[model_type],
+        )
+        session["model" + model_type] = model
+        session["vocab" + model_type] = vocab
```

Comment:
This function has become pretty long now. Maybe we can break it up into some smaller coherent functions. We could move this to some kind of store_model_in_session and store_vocab_in_session

Reply:
True, I also realized that the 'existing' dataset varies per model_type, so I should also fix that in the function.